proposed/local-sdk-global-json.md

```diff
@@ -0,0 +1,83 @@
+
+```json
+{
+    "sdk": {
+        "additionalPaths": [ ".dotnet" ],
+    }
+}
+```
+
+In this configuration the host resolver would find a compatible SDK if it exists in `.dotnet` or a machine wide location.
+
+The values in the `additionalLocations` property can be a relative or absolute path. When a relative path is used it will be resolved relative to the location of global.json. These values also support the following substitutions:
+
+- `"$local$"`: this matches the local installation point of .NET SDK for the current operating system: `%LocalAppData%\Microsoft\dotnet`` on Windows and `$HOME/.dotnet`` on Linux/macOS.
+- `"$machine$"`: this matches the machine installation point of .NET for the current operating system.
+- `%VARIABLE%/$VARIABLE`: environment variables will be substituted. Either the Windows or Unix format can be used here and will be normalized for the operating system the host resolver executes on.
+
+The host resolver will consider the `additionalLocations` in the order they are defined and will stop at the first match. If none of the locations have a matching SDK then it it will fall back to considering machine wide installations (unless `additionalLocationsOnly` is `true`).
+
+This design requires us to only change the host resolver. That means other tooling like Visual Studio, VS Code, MSBuild, etc ... would transparently benefit from this change. Repositories could update global.json to have `additionalLocations` support `.dotnet` and Visual Studio would automatically find it without any design changes.
+
+## Considerations
+### Installation Points
+One item to keep in mind when considering this area is the .NET SDK can be installed in many locations. The most common are:
+
+- Machine wide
+- User wide: `%LocalAppData%\Microsoft\dotnet`` on Windows and `$HOME/.dotnet`` on Linux/macOS.
+- Repo: `.dotnet`
+
+Our installation tooling tends to avoid redundant installations. For example if restoring a repository that requires 7.0.400, the tooling will not install it locally if 7.0.400 is installed machine wide. It also will not necessarily delete the local `.dotnet` folder or the user wide folder. That means developers end up with .NET SDK installs in all three locations but only the machine wide install has the correct ones.
+
+As a result solutions like "just use .dotnet if it exists" fall short. It will work in a lot of casse but will fail in more complex scenarios. To completely close the disconnect here we need to consider all the possible locations.
+
+### Do we need additionalLocationsOnly?
+The necessity of this property is questionable. The design includes it for completeness and understanding that the goal of some developers is complete isolation from machine state. As long as we're considering designs that embrace local deployment, it seemed sensible to extend the design to embrace _only_ local deployment.
```

Comment:
There are things that can change a particular SDK state (namely workload installs). So in the event a user has the same SDK in both the additional paths and globally, this may be needed for that type of situation but it's also likely fairly niche.

Reply:
Do I understand correctly this in effect equals to `DOTNET_MULTILEVEL_LOOKUP=0`?

Reply:
That is my understanding as well. With the difference that now it would work the same across all platforms (the env. variable only ever worked on windows, and now it's off by default anyway).

Reply:
> I don't like this property as much. If the goal of some developers would be isolation from the machine state, then why not just set DOTNET_ROOT after a repo-specific install?

How can tooling like say Visual Studio pick up such an environment variable? Yes you can do tricks like set the variable then start VS but that impacts the entire dev workflow. Actions like using the quick  launch menu don't work anymore cause VS can't process your repo without extra information. Any external process that launches VS (like say a debugger attach event) don't work in the repo cause they don't set the environment variable. It's an endless series of gotchas.